docs/visualizing-your-sql-queries/plotting-with-ggplot.md

```diff
@@ -1,14 +1,638 @@
+
+The `ggplot` API is built on top of `matplotlib` and is structured around the principles of the Grammar of Graphics, allowing you to build any graph using the same `ggplot2` components: a data set, a coordinate system, and geoms (geometric objects). However, to make the API suitable for JupySQL, we <b>input a SQL table name, instead of a dataset</b>. Therefore, the same [workflow](https://ploomber-sql.readthedocs.io/en/latest/visualizing-your-sql-queries/plotting-with-seaborn.html#barplots) of creating our CTE, which was employed in the `seaborn` tutorial, will be in action here as well (no need to convert the CTE into a pandas `DataFrame()`).
+
+<b>Note</b> at this point, JupySQL's `ggplot` API supports:
+
+Aes:
+
+- `x` - a SQL column mapping
+
+- `color` and `fill` to apply edgecolor and fill colors to plot shapes
+
+Geoms:
+
+- `geom_boxplot` similar to `%sqlplot boxplot`
+
+- `geom_histogram` similar to `%sqlplot histogram`
+
+Facet:
+
+- `facet_wrap` to display multiple plots in a single layout
+
+```{important}
+Lastly, unlike `%sqlplot` that returns a `matplotlib.Axes` object, the `ggplot` API returns a `ggplot` object, which cannot be customized like we did when using `%sqlplot`. Therefore, customizing axes labels and titles is not possible in `ggplot` API yet.
+```
+
+### `ggplot` Template
+
+To build a graph, we first should initialize a `ggplot` instance with a reference to our SQL table using the `table` parameter, and a mapping object. Here’s is the complete template to build any graph:
+
+```python
+(
+    ggplot(table='sql_table_name', mapping=aes(x='table_column_name'))
+    +
+    geom_func() # geom_histogram or geom_boxplot (required)
+    +
+    facet_func() # facet_wrap (optional)
+)
+```
+
+```{important}
+When working with CTE's, we must include it along with `table` using the `with_` parameter. In this tutorial, we will be using CTE's throughout our examples.
+```
+
+### `ggplot` CTE
+
+We will create a CTE to use throughout the `ggplot` examples. All the information from the three tables, `account`, `district`, and `loan`, will be present in this CTE, named `ggplot_CTE`, for convenience. The `--no-execute` function tells JupySQL to <i>skip the execution of the stored query</i>.
+
+```{code-cell} ipython3
+%%sql --save ggplot_CTE --no-execute
+SELECT *
+FROM s1.account AS a
+LEFT JOIN s1.loan AS l
+ON a.account_id = l.account_id
+LEFT JOIN s1.district AS d
+ON d.district_id = a.district_id
+```
+
+Now, let's look at different types of visualizations using the `ggplot` API and test ourselves on them!
+
+### `geom_boxplot`
+
+Suppose the finance manager now wants to obtain the distribution of demographic data and, hence, wants a boxplot of the number of municipalities that have less than 500 inhabitants and those between 500 and 2000 inhabitants across all regions in the data.
```

Comment:
I'm not so sure if the above CTE should be used to answer this hypothetical question. There may be duplicates of districts since we are using `LEFT JOIN` on the `s1.account` table. I think we  should just use the `s1.district` table instead to answer this question.

Using the CTE of the joined tables: 
![image](https://github.com/ploomber/sql/assets/96606221/74f11709-e716-4249-990f-29d6b859e828)

Using just `s1.district`
![image](https://github.com/ploomber/sql/assets/96606221/4a8754e0-4046-4482-be94-8341f296d628)

In this case, the mean of `no_of_municipalities_lt_499` is lower for the CTE of the joined tables. This could be because the `s1.accounts` table is duplicating districts with a lower count of `no_of_municipalities_lt_499`.  I could be wrong though, please let me know if this makes sense.

Reply:
Brilliant catch and agreed. I shall change the example to the same one that was in the `sqlplot boxplot` example (`payments` and `average salary` boxplot).

Reply:
Yes, please open an issue